src/rust/ide/view/src/window_control_buttons/common.rs

```diff
@@ -0,0 +1,325 @@
+    , icon             : AnyShape
+    , radius           : Var<Pixels>
+    ) -> AnyShape {
+        let background_color = Var::<Rgba>::from(background_color);
+        let icon_color       = Var::<Rgba>::from(icon_color);
+        let circle           = Circle(radius).fill(background_color);
+        let icon             = icon.fill(icon_color);
+        (circle + icon).into()
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+/// An internal model of Top Buttons Panel button component
+#[derive(Clone,CloneRef,Debug)]
+#[clone_ref(bound="Shape:CloneRef")]
+#[allow(missing_docs)]
+pub struct Model<Shape> {
+    app            : Application,
+    logger         : DefaultTraceLogger,
+    display_object : display::object::Instance,
+    shape          : ShapeView<Shape>,
+}
+
+impl<Shape: ButtonShape> Model<Shape> {
+    /// Construct a button's model.
+    pub fn new(app:&Application) -> Self {
+        let app    = app.clone_ref();
+        let logger = DefaultTraceLogger::new(Shape::debug_name());
+        let display_object  = display::object::Instance::new(&logger);
+        let shape  = ShapeView::new(&logger);
+        display_object.add_child(&shape);
+        Self{app,logger,display_object,shape}
+    }
+
+    /// Set the background (i.e. the circle) color.
+    pub fn set_background_color(&self, color:impl Into<Rgba>) {
+        self.shape.background_color().set(color.into().into());
+    }
+
+    /// Set the icon (i.e. the shape on the circle) color.
+    pub fn set_icon_color(&self, color:impl Into<Rgba>) {
+        self.shape.icon_color().set(color.into().into());
+    }
+
+    /// Retrieves circle radius value from an frp sampler event.
+    fn get_radius(radius:&Option<style::data::Data>) -> f32 {
+        radius.as_ref().and_then(DataMatch::number).unwrap_or(RADIUS_FALLBACK)
+    }
+
+    /// Set radius, updating the shape sizes and position.
+    pub fn set_radius(&self, radius:&Option<style::data::Data>) -> Vector2<f32> {
+        let radius = Self::get_radius(radius);
+        let size   = Self::size_for_radius(radius);
+        self.shape.size().set(size);
+        self.shape.set_position_x(radius);
+        self.shape.set_position_y(-radius);
+        size
+    }
+
+    /// Calculate the size of button for a given radius value.
+    pub fn size_for_radius(radius:f32) -> Vector2<f32> {
+        Vector2(radius,radius) * 2.0
+    }
+}
+
+
+
+// ===========
+// === FRP ===
+// ===========
+
+ensogl::define_endpoints! {
+    Input {
+        enabled (bool),
+        mouse_nearby (bool),
+    }
+    Output {
+        clicked (),
+        is_hovered (bool),
+        state(State),
+        size (Vector2<f32>),
+    }
+}
+
+
+
+// ============
+// === View ===
+// ============
+
+/// The StatusBar component view.
+///
+/// The status bar gathers information about events and processes occurring in the Application.
+// TODO: This is a stub. Extend it when doing https://github.com/enso-org/ide/issues/1193
```

Comment:
Well, this is not a status bar, right?

Remember to put here a description about "click" magic (to see clearly, that this button differs from toggle-one.